### PR TITLE
Cleanup the point lighting gbi implementation

### PR DESCRIPTION
**Pull request by Thar0**
This PR updates the gbi.h point lighting implementation to be more complete.
- Wraps all of the point lighting features in a macro to clearly identify it.
- Adds comments to the `PointLight_t` structure describing the fields.
- Adds `G_LIGHTING_POSITIONAL_H`, for completeness.
- Cleans up the `gdSPDefLights*` macros
- Adds `gdSPDefPointLights` macros, for completeness.

### CODE REVIEW

File: include/PR/gbi.h [hunk 1]
```diff
@@ -1277,15 +1285,16 @@ typedef struct {
...
-  unsigned char colc[3];
-  unsigned char kl;
-  short pos[3];
-  unsigned char kq;
+  unsigned char col[3];		/* point light value (rgba) */
+  unsigned char kc;         /* constant attenuation (> 0 indicates point light) */
+  unsigned char colc[3];	/* copy of point light value (rgba) */
+  unsigned char kl;         /* linear attenuation */
+  short pos[3];             /* world-space position of light */
+  unsigned char kq;         /* quadratic attenuation */
```
Comment (AngheloAlf) [inline include/PR/gbi.h:1295]: tabs :despair:

**Review by AngheloAlf — APPROVED** [review]

File: include/PR/gbi.h [hunk 2]
```diff
@@ -16,6 +16,9 @@
 
 #define F3DEX_GBI_2
 
+/* F3DEX2 with Point Lighting */
+#define F3DEX_GBI_PL
```
Comment (AngheloAlf) [inline include/PR/gbi.h:20]: Actually I wonder if we should pass this define (and `F3DEX_GBI_2`) in the Makefile instead of just defining it here

Comment (hensldm) [reply include/PR/gbi.h:20]: Would we want to auto define `F3DEX_GBI_2` if `F3DEX_GBI_PL` is passed, so we only had to pass `F3DEX_GBI_PL` on the makefile? Something like

```c
#ifdef F3DEX_GBI_PL
#define F3DEX_GBI_2
#endif
```

File: include/PR/gbi.h (hunk 1 above)
Comment (Thar0) [reply include/PR/gbi.h:1295]: The entire file needs reformatting so I didn't bother with it yet. I plan to follow-up with a formatting pass on gbi.h that makes it the same as OoT.

File: include/PR/gbi.h (hunk 2 above)
Comment (Thar0) [reply include/PR/gbi.h:20]: I think this is a good idea and is how it's supposed to work, but I'd like to do this separately. As mentioned above, I have plans for a formatting pass on this file and it seems like a good time to do this then.

**Review by hensldm — APPROVED** [review]

**Pull request merged**